Adjustable Pick Timer #2

dev-id opened the pull request.

arxanas committed:
Hackily add EMA

dev-id committed:
timer

dev-id committed:
timer

dev-id committed:
timer

dev-id committed:
timer

dev-id committed:
timer

dev-id committed:
timer

dev-id committed:
Timer syntax fix

dev-id committed:
Timer syntax fix

dev-id committed:
Sort lands separate from 0cmc nonlands

dev-id committed:
Sort lands separate from 0cmc nonlands

dev-id committed:
wtf syntax

dev-id committed:
wtf syntax

arxanas commented on public/src/components/game.js:
```diff
@@ -69,7 +69,12 @@ export default React.createClass({
       d.div({}, `Format: ${App.state.format}`),
       LBox('bots', 'bots'),
-      LBox('timer', 'timer'),
+      d.div({}, d.input({
```
There is a `num-lands` CSS class that shortens the textbox width. It might be a good idea to refactor that into a general-purpose number text box class and use it here, since the textbox is rather wide.

arxanas commented on public/src/components/game.js:
```diff
@@ -69,7 +69,12 @@ export default React.createClass({
+        max: 60,
+        type: 'number',
+        valueLink: App.link('timer'),
+        }), ' second timer'),
```
You should wrap the text and its `input` in a `label`, so that one can click the text and it focuses the input. It's also probably good for accessibility.

arxanas commented on src/human.js:
```diff
@@ -69,8 +69,8 @@ module.exports = class extends EventEmitter {
 
-    if (this.useTimer)
-      this.time = 20 + 5 * pack.length
+    if (this.useTimer > 0)
```
Is it apparent to the user that a value of zero is "no timer"? Maybe add some `title` text to the label text, or have both a checkbox that enables the time (and shows the timer dialog) or disables it. I might fix that myself if I get some time.

arxanas commented on src/human.js:
```diff
@@ -69,8 +69,8 @@ module.exports = class extends EventEmitter {
-    if (this.useTimer)
-      this.time = 20 + 5 * pack.length
+    if (this.useTimer > 0)
+      this.time = parseInt(this.useTimer) + parseInt(pack.length)
```
It's probably idiomatic to use `parseInt(foo, 10)` as per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt.

arxanas commented on src/make/cards.js:
```diff
@@ -275,13 +275,16 @@ function doCard(rawCard, cards, code, set) {
     name = rawCard.names.join(' // ')
+    
+  //separate landsfrom 0cmc cards by setting 0cmc to .2
+  var cmcadjusted = rawCard.cmc || 0.2
```
I think this should be camelCased to be `cmcAdjusted`.

For the values `0.2` and `0.1` perhaps they should be constants listed in the same place, because otherwise the reader will have to pan around the file to find related values. `0.2` suggests that there is a partner value but someone might not realize that `0.1` has a partner value.

arxanas commented on src/make/cards.js:
```diff
@@ -292,11 +295,15 @@ function doCard(rawCard, cards, code, set) {
     colors[0].toLowerCase()
+    
+  //set lands to .1 to sort them before nonland 0cmc  
+  if ('Land'.indexOf(rawCard.types) > -1)
```
Shouldn't this be `rawCard.types.indexOf('Land')`, or am I missing something? (Also, will the text `Land` be capitalized in the string? It might be prudent to just lowercase both for safety.)

arxanas committed:
Add Gitter and Waffle.io badges to README

arxanas committed:
Bugfix: Prevent crashes when sock.h is undefined

Occasionally there is a crash because sock.h is undefined when trying
to mark a player as disconnected. It's unclear to me why this is
happening, but theoretically this commit should prevent this situation
from happening as we should now always have a `.h` on hand.

arxanas committed:
Update pick length timer with feedback in PR #2

arxanas committed:
Revert "Sort lands separate from 0cmc nonlands"

This reverts commits 155fcb4bc496b00f470deac8bde532f69cfb60e2 and
a5a8aa6aa45d7cb3dc2e6fa927858b7520abd01c.

arxanas committed:
Use comparator function to sort lands before non-lands

arxanas committed:
Move branding strings into config. Closes #6

This adds a configuration file for both client and server. Page sections
(MOTD, footer) are now rendered on the client-side, which means no
orchestrating the dangerous rendering of HTML sent by the server.

You will need to generate `config.client.js` and `config.server.js`
before reloading the server.

arxanas committed:
Move branding strings into config. Closes #6

This adds a configuration file for both client and server. Page sections
(MOTD, footer) are now rendered on the client-side, which means no
orchestrating the dangerous rendering of HTML sent by the server.

You will need to generate `config.client.js` and `config.server.js`
before reloading the server.

arxanas committed:
Implement helper function to add spacer in between elements

arxanas committed:
Implement helper function to add spacer in between elements

arxanas committed:
Merge branch 'enhancement/config' into live

dev-id committed:
Arxanas changes

dev-id committed:
Merge pull request #6 from arxanas/live

Live

arxanas commented:
Merged the pick timer into master.
